app/uk/gov/hmrc/tariffclassificationfrontend/service/CasesService.scala

```diff
@@ -136,4 +137,29 @@ class CasesService @Inject()(appConfig: AppConfig, auditService: AuditService,
     }
   }
 
+  def addKeyword(c: Case, keyword: String)(implicit hc: HeaderCarrier): Future[Case] = {
+    c.keywords match {
+      case keywords if keywords.contains(keyword.toUpperCase) => Future.successful(c)
+      case _ =>
+        val caseToUpdate = c.copy(keywords = c.keywords + keyword.toUpperCase)
+        connector.updateCase(caseToUpdate)
+    }
+  }
+
+  def removeKeyword(c: Case, keyword: String)(implicit hc: HeaderCarrier): Future[Case] = {
+    c.keywords match {
+      case keywords if keywords.contains(keyword.toUpperCase) =>
+        val caseToUpdate = c.copy(keywords = c.keywords - keyword.toUpperCase)
+        connector.updateCase(caseToUpdate)
+      case _ => Future.successful(c)
+    }
+  }
+
+  def autoCompleteKeywords(): Future[Seq[String]] = {
+    Future {
+      val url = getClass.getClassLoader.getResource("keywords.txt")
```

Comment:
I would move this logic (load keywords from TXT file to memory) in a separate file (maybe `KeywordsService`)

Reply:
Done